be/src/util/buffered_stream.h

```diff
@@ -64,4 +71,41 @@ class BufferedInputStream {
     uint64_t _file_offset = 0;
 };
 
+class SharedBufferedInputStream : public IBufferedInputStream {
```

Comment:
Could you add some comments for this class?
What is the purpose and how does this class achieve the goal.

Reply:
OK.